fastly/block_fastly_service_v1_logging_ftp_test.go

```diff
@@ -0,0 +1,286 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	fst "github.com/fastly/go-fastly/fastly"
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestResourceFastlyFlattenFTP(t *testing.T) {
+	cases := []struct {
+		remote []*fst.FTP
+		local  []map[string]interface{}
+	}{
+		{
+			remote: []*fst.FTP{
+				{
+					Version:         1,
+					Name:            "ftp-endpoint",
+					Address:         "ftp.example.com",
+					Username:        "username",
+					Password:        "password",
+					PublicKey:       pgpPublicKey(),
+					Path:            "/path",
+					Port:            21,
+					Period:          3600,
+					GzipLevel:       3,
+					Format:          "%h %l %u %t \"%r\" %>s %b",
+					FormatVersion:   2,
+					TimestampFormat: "%Y-%m-%dT%H:%M:%S.000",
+					Placement:       "none",
+				},
+			},
+			local: []map[string]interface{}{
+				{
+					"name":             "ftp-endpoint",
+					"address":          "ftp.example.com",
+					"user":             "username",
+					"password":         "password",
+					"public_key":       pgpPublicKey(),
+					"path":             "/path",
+					"period":           uint(3600),
+					"port":             uint(21),
+					"gzip_level":       uint8(3),
+					"format_version":   uint(2),
+					"format":           "%h %l %u %t \"%r\" %>s %b",
+					"timestamp_format": "%Y-%m-%dT%H:%M:%S.000",
+					"placement":        "none",
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		out := flattenFTP(c.remote)
+		if diff := cmp.Diff(out, c.local); diff != "" {
+			t.Fatalf("Error matching: %s", diff)
+		}
+	}
+}
+
+func TestAccFastlyServiceV1_logging_ftp_basic(t *testing.T) {
+	var service fst.ServiceDetail
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	domain := fmt.Sprintf("fastly-test.%s.com", name)
+
+	log1 := fst.FTP{
+		Version:         1,
+		Name:            "ftp-endpoint",
+		Address:         "ftp.example.com",
+		Username:        "user",
+		Password:        "p@ssw0rd",
+		PublicKey:       pgpPublicKey() + "\n", // The '\n' is necessary becaue of the heredocs (i.e., EOF) in the config below.
+		Path:            "/path",
+		Port:            27,
+		GzipLevel:       3,
+		Period:          3600,
+		TimestampFormat: "%Y-%m-%dT%H:%M:%S.000",
+		Format:          "%h %l %u %t \"%r\" %>s %b",
+		FormatVersion:   2,
+		Placement:       "none",
+	}
+
+	log1_after_update := fst.FTP{
+		Version:         1,
+		Name:            "ftp-endpoint",
+		Address:         "ftp2.example.com",
+		Username:        "user",
+		Password:        "p@ssw0rd2",
+		PublicKey:       pgpPublicKey() + "\n", // The '\n' is necessary becaue of the heredocs (i.e., EOF) in the config below.
```

Comment:
Note this ugly `+ "\n"` and accompanying comment.

This is b/c of [this](https://github.com/terraform-providers/terraform-provider-fastly/pull/235/files#diff-16cf93bc3363904e6d40ed935a91db27R274).